src/wp-includes/class-wp-query.php

```diff
@@ -3476,7 +3476,18 @@ public function get_posts() {
 			$this->posts = array_map( 'get_post', $this->posts );
 
 			if ( $q['cache_results'] ) {
-				update_post_caches( $this->posts, $post_type, $q['update_post_term_cache'], $q['update_post_meta_cache'] );
+				$post_ids       = wp_list_pluck( $this->posts, 'ID' );
+				$non_cached_ids = _get_non_cached_ids( $post_ids, 'posts' );
+				if ( ! empty( $non_cached_ids ) ) {
+					$fresh_posts = array();
```

Comment:
I'm having trouble finding a way to pass this condition and enter this part of the code (testing on the main query).

Are you able to give me some instructions as to how to force that/patiently explain that that's the point? Looking at line 3476, I'd expect the caches to be warm due to the `get_post` calls.